query/src/test/scala/filodb/query/exec/AggrOverRangeVectorsSpec.scala

```diff
@@ -310,6 +320,38 @@ class AggrOverRangeVectorsSpec extends FunSpec with Matchers with ScalaFutures {
     result4(0).key shouldEqual noKey
     compareIter(result4(0).rows.map(_.getDouble(1)), Seq(Double.NaN, 5.133333333333333d).iterator)
 
+    // BottomK
+    val resultObs5a = RangeVectorAggregator.mapReduce(AggregationOperator.BottomK,
+      Seq(2.0), false, Observable.fromIterable(samples), noGrouping)
+    val resultObs5 = RangeVectorAggregator.mapReduce(AggregationOperator.BottomK,
+      Seq(2.0), true, resultObs5a, rv=>rv.key)
+    val resultObs5b = RangeVectorAggregator.present(AggregationOperator.BottomK, Seq(2.0), resultObs5, 1000)
+    val result5 = resultObs5.toListL.runAsync.futureValue
+    result5.size shouldEqual 1
+    result5(0).key shouldEqual noKey
+    compareIter2(result5(0).rows.map(r=> Set(r.getDouble(2), r.getDouble(4))),
```

Comment:
Lets add more documentation on what we are validating, especially since the operations are a bit complex.

Reply:
Added comments